.github/workflows/main.yml

```diff
@@ -0,0 +1,51 @@
+
+name: main
+
+on:
+  pull_request:
+    branches: ['*']
+  push:
+    branches: ['*']
+
+concurrency: ci-${{ github.ref }}
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Static Analysis
+        uses: pre-commit/action@v3.0.0
+
+      - name: cdk synth
+        uses: youyo/aws-cdk-github-actions@v2
+        with:
+          cdk_subcommand: 'synth'
+          actions_comment: false
+          cdk_args: '--context env=dev --debug'
+
+      - name: Install dependencies
+        run: pip install -r requirements.txt -r requirements-dev.txt
+
+      - name: Run unit tests
+        run: python -m pytest tests/ -s -v
+
+  cdk-deploy-dev:
+    if: ${{github.ref == 'refs/heads/main'}}
```

Comment:
shouldn't this be `refs/heads/dev` to match the context name?

Reply:
Yes, that's cleaner.  I changed the branch name from main to dev and then updated the "if" statement.